components/tikv_util/src/sys/thread.rs

```diff
@@ -264,10 +381,43 @@ mod tests {
     }
 
     #[test]
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
     fn test_thread_ids() {
-        let ids: std::io::Result<Vec<_>> = thread_ids(process_id());
-        assert!(ids.is_ok());
-        assert!(!ids.unwrap().is_empty());
+        let actual_tids = Arc::new(Mutex::new(vec![]));
+        let stop_threads = Arc::new(AtomicBool::new(false));
+
+        let threads = (0..10)
+            .map(|_| {
+                let actual_tids = actual_tids.clone();
+                let stop_threads = stop_threads.clone();
+                std::thread::spawn(move || {
+                    {
+                        actual_tids.lock().unwrap().push(thread_id());
+                    }
+                    while !stop_threads.load(Ordering::Relaxed) {}
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let ids: HashSet<_> = thread_ids(process_id()).unwrap();
+        {
+            let actual_tids = actual_tids.lock().unwrap();
+            for tid in &*actual_tids {
+                assert!(ids.contains(tid));
```

Comment:
Race with L395, should use channel instead.

Reply:
updated